version: "3.9"

services:
  hw26:
    build:
      context: .
    image: $IMAGE_TAG:$GITHUB_REF_NAME-$GITHUB_RUN_ID
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    ports:
      - 80:80
    volumes:
      - ./app_config.py:/code/default_config.py
    depends_on:
      pg:
        condition: service_healthy
  pg:
    image: postgres:latest
    environment:
      POSTGRES_USER: ${POSTGRES_USER}
      POSTGRES_PASSWORD: ${POSTGRES_PASSWORD}
      POSTGRES_DB: ${POSTGRES_DB}
    volumes:
      - ./hw26_data:/var/lib/postgresql/data 700
    healthcheck:
      test: [ "CMD-SHELL", "pg_isready -U postgres" ]
      interval: 5s
      timeout: 5s
      retries: 5
